### ISSUE TITLE
Missing `ur_device_info_t` enum values

### ISSUE THREAD

**Issue by callumfare** [oneapi-src/unified-runtime#374]
There is no equivalent for the following PI enums in UR:
```
PI_DEVICE_INFO_BUILD_ON_SUBDEVICE
PI_EXT_ONEAPI_DEVICE_INFO_MAX_WORK_GROUPS_3D
PI_EXT_INTEL_DEVICE_INFO_MEMORY_BUS_WIDTH
PI_EXT_ONEAPI_DEVICE_INFO_CUDA_ASYNC_BARRIER
```

It's not clear (to me anyway) whether UR has exact equivalents for these:
```
PI_DEVICE_INFO_GPU_HW_THREADS_PER_EU
PI_DEVICE_INFO_GPU_EU_COUNT_PER_SUBSLICE
```

Ideally we need equivalents so pi2ur can map the device info properly. #55 covered this, but the PR that resolved this doesn't seem to have covered all the missing values.

**Comment by kbenzie** [comment]
https://github.com/intel/llvm/pull/9203 will also add `PI_EXT_INTEL_DEVICE_INFO_MEM_CHANNEL_SUPPORT` or similar.

**Comment by kbenzie** [comment]
Fixed by #478.
